src/js/csv.js

```diff
@@ -1,19 +1,48 @@
-/* Copyright(c) 2017 Philip Mulcahy. */
+/* Copyright(c) 2017-2020 Philip Mulcahy. */
 /* jshint strict: true, esversion: 6 */
+/* jslint node:true */
 
 "use strict";
 
 import save_file from './save_file';
 
-function download(table) {
+function download(table, type) {
     const tableToArrayOfArrays = function(table) {
         const rows = table.rows;
         const result = [];
-        for (let i=0; i<rows.length; ++i) {
-            const cells = rows[i].cells;
-            const cell_array = [];
-            for (let j=0; j<cells.length; ++j) {
-                cell_array.push(cells[j].textContent);
+        for(let i=0; i<rows.length + ( type  ?  -1  :  0 ); ++i) {
+            let cells = rows[i].cells;
```

Comment:
what's with replacing all the consts? - I'm pretty sure (not money sure) that this one could have stayed const.
Const is a good thing - use it if you possibly can, and only upgrade to a mutable if you absolutely have to - it makes it easier to reason about code (particularly "big" code like this), and reduces the chances for hard to fix bugs.

Reply:
I checked and they're not changed after creation, so I changed them to consts.